app/Notifications/SlackTagNotification.php

```diff
@@ -0,0 +1,73 @@
+<?php
+
+namespace Rogue\Notifications;
+
+use Rogue\Models\Post;
+use Illuminate\Bus\Queueable;
+use Illuminate\Notifications\Notification;
+use Illuminate\Notifications\Messages\SlackMessage;
+
+class SlackTagNotification extends Notification
+{
+    use Queueable;
+
+    public $post;
+
+    /**
+     * Create a new notification instance.
+     *
+     * @return void
+     */
+    public function __construct(Post $post)
+    {
+        $this->post = $post;
+    }
+
+    /**
+     * Get the notification's delivery channels.
+     *
+     * @param  mixed  $notifiable
+     * @return array
+     */
+    public function via($notifiable)
+    {
+        return ['slack'];
+    }
+
+    /**
+     * Get the mail representation of the notification.
+     *
+     * @param  mixed  $notifiable
+     * @return \Illuminate\Notifications\Messages\MailMessage
+     */
+    public function toSlack($notifiable)
+    {
+        if (config('services.slack.url')) {
```

Comment:
Nitpicky, but it could be cleaner to reformat this as a [guard clause](https://refactoring.com/catalog/replaceNestedConditionalWithGuardClauses.html):

```php
if (! config('services.slack.url')) {
    return;
}

return (new SlackMessage)
    ->from('Rogue', ':tonguecat:')
    // ...
```